### PR TITLE
crane 0.4.0 (new formula)

### LINKED ISSUE #914: Add crane to homebrew?
My limited understanding of homebrew is that you can either add a formula to homebrew/core or maintain a tap.

It seems like less over head if we can get into core (we don't have to maintain a separate repo), but from [their guidelines](https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff):

> We will reject formulae that seem too obscure, partly because they won’t get maintained and partly because we have to draw the line somewhere.
> 
> We frown on authors submitting their own work unless it is very popular.

I'm not sure if `[g]crane` is "very popular", but it seems like it's more likely to be accepted if _someone else_ submits it...

Also, we need a homepage.

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

closes https://github.com/google/go-containerregistry/issues/914

### CODE REVIEW

**Review by SMillerDev — APPROVED** [review]

File: Formula/crane.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "-ldflags",
+      "-s -w -X github.com/google/go-containerregistry/cmd/crane/cmd.Version=#{version}",
+      "./cmd/crane"
+  end
+
+  test do
+    json_output = shell_output("#{bin}/crane manifest homebrew/brew")
```
Comment (ImJasonH) [inline Formula/crane.rb:17]: `crane manifest gcr.io/google_containers/pause` ?

Comment (jonjohnsonjr) [reply Formula/crane.rb:17]: Would rather use `google-containers` -- the underscore thing is a historical curiosity.

Though, really, we should probably use something under the control of the project, e.g. `gcr.io/go-containerregistry/crane`?

Comment (carlocab) [reply Formula/crane.rb:17]: Thanks for looking at the test. Does either of you mind opening a PR for it?

Comment (jonjohnsonjr) [reply Formula/crane.rb:17]: I'll send a PR.

**Pull request closed without merging**